libraries/helpers.rb

```diff
@@ -19,25 +19,35 @@
 
 module StrongDM
   module Helpers
-    def sdm_relay_token(type = 'relay')
+    def sdm_relay_token(admin_token, name, type = 'relay', ip = nil, port = 5000, bind_ip = '0.0.0.0', bind_port = 5000)
+      return nil if admin_token.nil?
+      ip = node['ipaddress'] if ip.nil?
+      name = node['fqdn'] if name.nil?
       token = Mixlib::ShellOut.new(
         sdm,
         'relay',
         type == 'gateway' ? 'create-gateway' : 'create',
         '--name',
-        node['fqdn'],
-        "#{node['strongdm']["#{type}_advertise_address"]}:#{node['strongdm']["#{type}_port"]}",
-        "#{node['strongdm']["#{type}_bind_address"]}:#{node['strongdm']["#{type}_bind_port"]}",
-        'env' => {
-          'SDM_ADMIN_TOKEN' => node['strongdm']['admin_token'],
-        }
+        name,
+        "#{ip}:#{port}",
+        "#{bind_ip}:#{bind_port}",
+        'environment' => { 'SDM_ADMIN_TOKEN' => admin_token }
```

Comment:
Was it intended to change this from `env` to `environment`?

Reply:
The documentation says `environment` and playing with it in `irb` confirms.

```
$ /opt/chefdk/embedded/bin/irb 
irb(main):001:0> require 'mixlib/shellout'
=> true
irb(main):002:0> Mixlib::ShellOut.new('echo hello')
=> <Mixlib::ShellOut#12395540: command: 'echo hello' process_status: nil stdout: '' stderr: '' child_pid: nil environment: {} timeout: 600 user:  group:  working_dir:  >
```

While `env` appeared to work, I have no idea what magic and dragons made it do so.